src/js-marketing-notification.ts

```diff
@@ -29,18 +36,35 @@ export class JsMarketingNotification {
       document.body.removeChild(this.notificationContainer);
   }
 
+  public set setOpenCallback(fn: Function) {
+    if (null == fn) return;
```

Comment:
One possibility here might be to do a `if (typeof(fn) !== 'function')` type of check here.